docs/standard/serialization/binaryformatter-migration-guide/choose-a-serializer.md

```diff
@@ -0,0 +1,75 @@
+| Serializing non-public types                   | ✔️              | ✔️                      | ✔️                    | ✔️ (resolver required)   | ✔️                               |
+| Serializing fields                             | ✔️              | ✔️ (opt in)             | ✔️                    | ✔️ (attribute required)  | ✔️ (attribute required)          |
+| Serializing non-public fields                  | ✔️              | ✔️ (resolver required)  | ✔️                    | ✔️ (resolver required)   | ✔️ (attribute required)          |
+| Serializing properties                         | ✔️<sup>*</sup>  | ✔️                      | ✔️                    | ✔️ (attribute required)  | ✔️ (attribute required)          |
+| Deserializing readonly members                 | ✔️              | ✔️ (attribute required) | ✔️                    | ✔️                       | ✔️ (parameterless ctor required) |
+| Polymorphic type hierarchy                     | ✔️              | ✔️ (attribute required) | ✔️                    | ✔️ (attribute required)  | ✔️ (attribute required)          |
+| AOT support                                    | ❌️              | ✔️                      | ❌                    | ✔️                       | ❌ (planned)                     |
+
+## JSON using System.Text.Json
+
+The [`System.Text.Json`](../system-text-json/overview.md) library is a modern serializer that emphasizes security, high performance, and low memory allocation for the JavaScript Object Notation (JSON) format. JSON is human-readable and has broad cross-platform support. While text-based format is not as compact as binary formats, it can be significantly reduced in size through compression.
+
+Serialization excludes non-public and readonly members unless specifically handled through attributes and constructors. System.Text.Json also supports [custom serialization and deserialization](../system-text-json/custom-contracts.md) for more control over how types are converted into JSON and vice versa. System.Text.Json does not support the `[Serializable]` attribute.
+
+[Migrate to System.Text.Json](./migrate-to-system-text-json.md).
+
+## XML using System.Runtime.Serialization.DataContractSerializer
+
+`DataContractSerializer` was introduced in .NET Framework 3.0 and is used to serialize and deserialize data sent in Windows Communication Foundation (WCF) messages. `DataContractSerializer` is an XML serializer that **fully supports the serialization programming model that was used by the `BinaryFormatter`**. It requires the known types to be specified up-front (but most .NET collections and primitive types are on a default allow-list and don't need to be specified). It's the serializer that requires the least amount of effort to migrate to.
+
+While `DataContractSerializer` carries those functional benefits when migrating from BinaryFormatter, it is not as performant as the other choices, nor does it use a cross-platform format.
```

Comment:
> nor does it use a cross-platform format.

What do you mean by that? I'd consider XML just as cross-platform as JSON.

Reply:
I was trying to indicate that there aren't cross-platform serialization libraries that produce/consume DataContractSerializer's XML structure. But you're right that the _format_ is cross-platform despite that. We could refine the table to capture this more accurately too.

Reply:
Oh you mean that the particular shape DCS produces isn't as customizable as JSON is with the JSON serializer? I guess that's fair; but JSON also has features (e.g. the reference graph) where the resulting JSON has a particular shape. Might be worth point out or not.

It just felt odd to say "XML isn't cross platform" :-)

Reply:
Makes sense. Thanks for taking the suggestion!